### PR TITLE
[CARBONDATA-1310]Merge test code for AddColumnTestCases, DropColumnTestCases and ChangeDataTypeTestCases

### PR DESCRIPTION
**Pull request by mayunSaicmotor**
merge test code for AddColumnTestCases, DropColumnTestCases and ChangeDataTypeTestCases
between packages,

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3118/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/527/

File: integration/spark2/src/test/scala/org/apache/spark/carbondata/restructure/vectorreader/AddColumnTestCases.scala [hunk 1]
```diff
@@ -42,26 +43,55 @@ class AddColumnTestCases extends Spark2QueryTest with BeforeAndAfterAll {
       "('DICTIONARY_EXCLUDE'='charField', 'DEFAULT.VALUE.charfield'='def')")
     sql(s"LOAD DATA LOCAL INPATH '$resourcesPath/restructure/data1.csv' INTO TABLE addcolumntest " +
         s"options('FILEHEADER'='intField,stringField,charField,timestampField,decimalField')")
+    sql("CREATE TABLE hivetable stored as parquet select * from addcolumntest")
   }
 
   test("test like query on new column") {
+    sqlContext.setConf("carbon.enable.vector.reader", "true")
```
Comment (chenliang613) [inline integration/spark2/src/test/scala/org/apache/spark/carbondata/restructure/vectorreader/AddColumnTestCases.scala]: Why need to add the vector settings?

Comment (mayunSaicmotor) [inline integration/spark2/src/test/scala/org/apache/spark/carbondata/restructure/vectorreader/AddColumnTestCases.scala]: true should means vector reader, false means row reader, actually it is the main difference between the 2 test class packages which need to merge

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3128/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/537/

**Comment by chenliang613** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3167/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/572/

File: integration/spark2/src/test/scala/org/apache/spark/carbondata/restructure/vectorreader/AddColumnTestCases.scala [hunk 2]
```diff
@@ -42,91 +43,263 @@ class AddColumnTestCases extends Spark2QueryTest with BeforeAndAfterAll {
...
-      "Alter table addcolumntest add columns(intnewField int) TBLPROPERTIES" +
-      "('DICTIONARY_INCLUDE'='intnewField', 'DEFAULT.VALUE.intNewField'='5')")
-    checkAnswer(sql("select charField from addcolumntest where intnewField > 2"),
-      Seq(Row("abc"), Row("def")))
-    checkAnswer(sql("select charField from addcolumntest where intnewField < 2"), Seq())
+    sqlContext.setConf("carbon.enable.vector.reader", "false")
+    checkAnswer(sql("select charField from addcolumntest where charField = 'abc'"), Row("abc"))
   }
 
+  test("test add dictionary column and test greaterthan/lessthan filter on new column") ({
```
Comment (chenliang613) [inline integration/spark2/src/test/scala/org/apache/spark/carbondata/restructure/vectorreader/AddColumnTestCases.scala]: why add "(" ?

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3171/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/577/

**Comment by chenliang613** [comment]
LGTM

**Pull request closed without merging**